src/idl_gen_swift.cpp

```diff
@@ -0,0 +1,721 @@
+  }
+
+  bool generate() {
+    namespace_depth = 0;
+    code_.Clear();
+    code_.SetValue("ACCESS", "_accessor");
+    code_ += "// " + std::string(FlatBuffersGeneratedWarning()) + "\n";
+    code_ += "import FlatBuffers\n";
+    // Generate code for all the enum declarations.
+
+    for (auto it = parser_.enums_.vec.begin(); it != parser_.enums_.vec.end();
+         ++it) {
+      const auto &enum_def = **it;
+      if (!enum_def.generated) {
+        SetNameSpace(enum_def.defined_namespace);
+        GenEnum(enum_def);
+      }
+    }
+
+    for (auto it = parser_.structs_.vec.begin();
+         it != parser_.structs_.vec.end(); ++it) {
+      const auto &struct_def = **it;
+      if (struct_def.fixed && !struct_def.generated) {
+        SetNameSpace(struct_def.defined_namespace);
+        GenStructReader(struct_def);
+      }
+    }
+
+    for (auto it = parser_.structs_.vec.begin();
+         it != parser_.structs_.vec.end(); ++it) {
+      const auto &struct_def = **it;
+      if (struct_def.fixed && !struct_def.generated) {
+        SetNameSpace(struct_def.defined_namespace);
+        GenStructWriter(struct_def);
+      }
+    }
+
+    for (auto it = parser_.structs_.vec.begin();
+         it != parser_.structs_.vec.end(); ++it) {
+      const auto &struct_def = **it;
+      if (!struct_def.fixed && !struct_def.generated) {
+        SetNameSpace(struct_def.defined_namespace);
+        GenTable(struct_def);
+      }
+    }
+
+    if (cur_name_space_) SetNameSpace(nullptr);
+
+    const auto filename = GeneratedFileName(path_, file_name_);
+    const auto final_code = code_.ToString();
+    return SaveFile(filename.c_str(), final_code, false);
+  }
+
+  void mark(const std::string str) {
+    code_.SetValue("MARKVALUE", str);
+    code_ += "\n// MARK: - {{MARKVALUE}}\n";
+  }
+
+  // Generates the create function for swift
+  void GenStructWriter(const StructDef &struct_def) {
+    code_.SetValue("STRUCTNAME", Name(struct_def));
+    std::string static_type = this->namespace_depth == 0 ? "" : "static ";
+    code_ += "public " + static_type + "func create{{STRUCTNAME}}(\\";
+    std::string func_header = "";
+    GenerateStructArgs(struct_def, &func_header, "");
+    code_ += func_header.substr(0, func_header.size() - 2) + "\\";
+    code_ += ") -> UnsafeMutableRawPointer {";
+    code_ +=
+        "\tlet memory = UnsafeMutableRawPointer.allocate(byteCount: "
+        "{{STRUCTNAME}}.size, alignment: {{STRUCTNAME}}.alignment)";
+    code_ +=
+        "\tmemory.initializeMemory(as: UInt8.self, repeating: 0, count: "
+        "{{STRUCTNAME}}.size)";
+    std::string func_body = "";
+    GenerateStructBody(struct_def, &func_body, "");
+    code_ += func_body + "\\";
+    code_ += "\treturn memory";
+    code_ += "}\n";
+  }
+
+  void GenerateStructBody(const StructDef &struct_def, std::string *code_ptr,
+                          std::string nameprefix, int offset = 0) {
+    std::string &code = *code_ptr;
```

Comment:
`code` used only once at line 133. is it really needed?

Reply:
well, I kinda stole the idea from csharp since the code is generated recursively. I did try removing it however that didn't work out as expected. if you have better ideas I am all ears.

Reply:
just realized we can directly use `code_`